mbed_host_tests/host_tests_conn_proxy/conn_proxy.py

```diff
@@ -28,7 +28,7 @@
 class KiViBufferWalker():
     """! Simple auxiliary class used to walk through a buffer and search for KV tokens """
     def __init__(self):
-        self.KIVI_REGEX = r"\{\{([\w\d_-]+);([^\}]+)\}\}\n"
```

Comment:
@bridadan in #104 your have add ```\n``` to the K,V regex. I think K,V protocol and ```\n``` and ```\r``` are different things. If htrun should send after receiving a ```\n``` then we should received until ```\n``` and then do the parsing. So that K,V parsing is independent of these terminating characters and also since data in K,V has nothing to do with serial limitation. In future this protocol can be on network interface as well.

So the work around for preventing DAPLINK bad state should be in the place where we are receiving data i.e. receive until ```\n```.